src/main/webapp/view/discussion/discussion-comment.js

```diff
@@ -0,0 +1,146 @@
+
+  static #SLOT_HEADER = 'header';
+  static #SLOT_CONTENT = 'content';
+  static #SLOT_REPLIES = 'replies';
+
+  static #SELECTOR_SHOW_REPLY = '#show-reply';
+  static #SELECTOR_REPLY_FORM = '#reply-form';
+  static #SELECTOR_CANCEL_REPLY = '#cancel-reply';
+  static #SELECTOR_POST_REPLY = '#post-reply';
+  static #SELECTOR_REPLY_TEXTAREA = '#reply-textarea';
+
+  #discussion;
+
+  /**
+   * Creates an custom HTML element representing a comment.  This uses the
+   * template and slots defined by `TEMPLATE` to render the
+   * comment's content and replies, once the comment is set.
+   *
+   * @param {DiscussionArea} discussion The current discussion.
+   */
+  constructor(discussion) {
+    super();
+    this.#discussion = discussion;
+    this.attachShadow({mode: 'open'});
+    const shadow = DiscussionComment.#TEMPLATE.content.cloneNode(true);
+    this.shadowRoot.appendChild(shadow);
+    this.addReplyEventListeners();
+  }
+
+  /**
+   * @param comment The comment from the discussion that this element should
+   *     render.
+   */
+  setComment(comment) {
+    this.comment = comment;
+    this.textContent = '';
+    this.setSlotSpan(
+        DiscussionComment.#SLOT_HEADER, this.getHeaderString(comment));
+    this.setSlotSpan(DiscussionComment.#SLOT_CONTENT, comment.content);
+    this.addReplies(comment.replies);
+  }
+
+  /**
+   * Returns a string containing the timestamp, author, and creation time of
+   * the `comment`.  The timestamp is not displayed for replies to
+   * other comments.
+   */
+  getHeaderString(comment) {
+    const username = comment.author.email.split('@')[0];
+    let timestampPrefix = '';
+    if (comment.type !== COMMENT_TYPE_REPLY) {
+      // Don't show timestamp on replies.
+      timestampPrefix = `${timestampToString(comment.timestampMs.value)} - `;
+    }
+    return `${timestampPrefix}${username} on ${comment.created}`;
+  }
+
+  /**
+   * Adds event listeners and handlers for the reply button and form.
+   * This makes the form open when "Reply" is pressed, submit when
+   * "Post" is pressed, and close when "Cancel" is pressed.
+   */
+  addReplyEventListeners() {
+    // The linter does not agree with the prettier about how to indent this.
+    /* eslint-disable indent */
+    const replyForm =
+        this.shadowRoot.querySelector(DiscussionComment.#SELECTOR_REPLY_FORM);
+    this.shadowRoot.querySelector(DiscussionComment.#SELECTOR_SHOW_REPLY)
+        .onclick = () => {
+      $(replyForm).collapse('show');
+    };
+    this.shadowRoot.querySelector(DiscussionComment.#SELECTOR_CANCEL_REPLY)
+        .onclick = () => {
+      $(replyForm).collapse('hide');
+    };
+    this.shadowRoot.querySelector(DiscussionComment.#SELECTOR_POST_REPLY)
+        .onclick = () => {
+      const textarea = this.shadowRoot.querySelector(
+          DiscussionComment.#SELECTOR_REPLY_TEXTAREA);
+      this.#discussion.postReply(textarea.value, this.comment.commentKey.id);
```

Comment:
Does this need to be in an ```initialize()``` function too?

Reply:
I created a new function for this event handler, lmk if it needs anything else